### ISSUE TITLE
new memory leak since v0.2.6

### ISSUE THREAD

**Issue by not-implemented** [mscdex/node-mariasql#144]
Since v0.2.6 we experience a bigger memory leak in production - we tracked things down to commit
3dbb651747a43246d5dcc0c8f29a1447352f076e.

We tried all the commits for a while in production (all tests with same Node version 4.4.3):

![memleak](https://cloud.githubusercontent.com/assets/1215371/15013697/8191ef48-1201-11e6-9169-b360c8fec2f4.png)

With commit 3dbb651747a43246d5dcc0c8f29a1447352f076e (not in the screenshot) the leak is already there.

We assume the leak occurs when the connection is closed by the server (we have a very small wait_timeout of 60 sec) ... but not sure. @devzer01 maybe you can have a look at it?

**Comment by devzer01** [comment]
Yeah it seems now that the leak in c bindings is exposed after fixing javascript, thanks for the issue report. We also experienced the issue after upgrading to 0.2.6

**Comment by devzer01** [comment]
Will take a look at the c code and push a PR

**Comment by not-implemented** [comment]
@devzer01 did you already find a solution here, or the cause of problem? We reviewed your changes recently, but didn't really find the reason for the memleak.

**Comment by devzer01** [comment]
The issues is in c bindings, because it's not releasing the handler from c
side

Sent via mobie
